add-device returns This object does not support toString() error

Executing the following command on an existing group (named sp_test):

```bash
java -jar AwsGreengrassProvisioner.jar -g sp_test --update-group --add-device test-device-meter
```

returns the following error:
```bash

[INFO] BasicGroupUpdateHelper: No thing ARN specified for device [test-device-meter], will re-use keys if possible
[INFO] BasicIotHelper: - Creating new keys.
[INFO] BasicIotHelper: Keys and certificate data written to [credentials/sp_test/test-device-meter.createKeysAndCertificate.serialized]
[ERROR] AwsGreengrassProvisioner: This object does not support toString()
java.lang.RuntimeException: This object does not support toString()
        at com.awslabs.data.NoToString.toString(NoToString.java:7)
        at java.lang.String.valueOf(String.java:2994)
        at java.lang.StringBuilder.append(StringBuilder.java:131)
        at com.awslabs.aws.greengrass.provisioner.implementations.helpers.BasicGreengrassHelper.getDevice(BasicGreengrassHelper.java:891)
        at com.awslabs.aws.greengrass.provisioner.implementations.helpers.BasicGroupUpdateHelper.addDevice(BasicGroupUpdateHelper.java:193)
        at com.awslabs.aws.greengrass.provisioner.implementations.helpers.BasicGroupUpdateHelper.execute(BasicGroupUpdateHelper.java:86)
        at com.awslabs.aws.greengrass.provisioner.implementations.helpers.BasicGroupUpdateHelper.execute(BasicGroupUpdateHelper.java:25)
        at com.awslabs.aws.greengrass.provisioner.interfaces.helpers.Operation.executeOrDisplayHelp(Operation.java:33)
        at io.vavr.control.Try.onSuccess(Try.java:712)
        at com.awslabs.aws.greengrass.provisioner.interfaces.helpers.Operation.execute(Operation.java:21)
        at com.awslabs.aws.greengrass.provisioner.AwsGreengrassProvisioner.lambda$null$3(AwsGreengrassProvisioner.java:81)
        at java.util.Optional.map(Optional.java:215)
        at com.awslabs.aws.greengrass.provisioner.AwsGreengrassProvisioner.lambda$run$da6d192b$1(AwsGreengrassProvisioner.java:81)
        at io.vavr.control.Try.of(Try.java:75)
        at com.awslabs.aws.greengrass.provisioner.AwsGreengrassProvisioner.run(AwsGreengrassProvisioner.java:76)
        at com.awslabs.aws.greengrass.provisioner.AwsGreengrassProvisioner.runProvisioner(AwsGreengrassProvisioner.java:56)
        at com.awslabs.aws.greengrass.provisioner.AwsGreengrassProvisioner.lambda$main$0(AwsGreengrassProvisioner.java:29)
        at io.vavr.control.Try.run(Try.java:118)
        at com.awslabs.aws.greengrass.provisioner.AwsGreengrassProvisioner.main(AwsGreengrassProvisioner.java:29)
[ERROR] AwsGreengrassProvisioner: This object does not support toString()
```

--- a/src/main/java/com/awslabs/aws/greengrass/provisioner/implementations/helpers/BasicGreengrassHelper.java
+++ b/src/main/java/com/awslabs/aws/greengrass/provisioner/implementations/helpers/BasicGreengrassHelper.java
@@ -888,7 +888,7 @@ public class BasicGreengrassHelper implements GreengrassHelper {
         }
 
         if (thingPrincipals.size() != 1) {
-            throw new RuntimeException("More than one principal found for [" + thingName + "], can not continue");
+            throw new RuntimeException(String.join("More than one principal found for [", thingName.getName(), "], can not continue"));
         }
 
         String certificateArn = thingPrincipals.get(0).getPrincipal();

